sqlest/src/main/scala/sqlest/ast/ColumnType.scala

```diff
@@ -96,15 +96,33 @@ object MappedColumnType {
    * Convenience method for constructing a `MappedColumnType` from a pair of
    * bidirectional mapping functions and an implicitly provided `BaseColumnType`.
    */
-  def apply[A, B: BaseColumnType](r: Option[B] => Option[A], w: A => B)(implicit innerColumnType: ColumnType.Aux[B, B]) = new MappedColumnType[A, B] {
+  def apply[A, B: BaseColumnType](r: B => A, w: A => B)(implicit innerColumnType: ColumnType.Aux[B, B]) = new MappedColumnType[A, B] {
     val baseColumnType = implicitly[BaseColumnType[B]]
-    def read(database: Option[Database]) = r(innerColumnType.read(database))
+    def read(database: Option[Database]) = r.lift(innerColumnType.read(database))
     def write(value: A) = innerColumnType.write(w(value))
   }
 
-  implicit class MappedColumnTypeOps[A, B](mappedColumnType: MappedColumnType[A, B]) {
-    def compose[C: BaseColumnType](inner: ColumnType.Aux[B, C]): MappedColumnType[A, C] = {
-      MappedColumnType((database: Option[C]) => mappedColumnType.read(inner.read(database)), (value: A) => inner.write(mappedColumnType.write(value)))
+  implicit class MappedColumnTypeOps[B, C: BaseColumnType](baseColumnType: ColumnType.Aux[B, C]) {
+    def compose[A](outerColumnType: ColumnType.Aux[A, B]): MappedColumnType[A, C] = {
+      val baseRead = (baseColumnType.read _).unlift
+      val mappedRead = (outerColumnType.read _).unlift
+      MappedColumnType(
+        (database: C) => mappedRead(baseRead(database)),
```

Comment:
I thinks it's better to just inline the definition here instead of having the UnliftOps
```scala
        (database: C) => outerColumnType.read(baseColumnType.read(Some(database))).get,

```